File: src/main/java/com/andy/Hello.java
package com.andy;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello world");
//        new Person().hello();
        String s = "abcde";
        System.out.println(s.charAt(3));
        //取多少字串長度
        System.out.println(s.length());
        //取到三位數之後
        System.out.println(s.substring(3));
        //物件 = 參照資料型態 通常宣告名稱開頭是大寫
        Person p = new Person("Andy",66.5f,1.7f);
//        p.hello();
        p.weight = 66.5f;
        p.height = 1.7f;
        System.out.println(p.bmi());
        int score = 88;
        System.out.println(score > 80 && score < 90);
        System.out.println(score < 80 || score > 90);
        char c = 'A';
        System.out.println(c > 60);

//        int age = 19;
//        Integer age2 = 19; //代表放在物件當中
//        char c = '我';
//        Character c2 = 'A';
//        byte b = 120;
//        Byte b2 = 120;
//        float weight = 66.5f;
//        double height = 1.7;
//        boolean adult = true;
//        boolean enroll = false;
//        String name = "andy";
//        String s = new String("ABCDE");
//        System.out.println(enroll);
//        System.out.println(s);
    }
}

File: src/main/kotlin/com/andy/Hello.kt
package com.andy

fun main() {
//    println("Hello kotlin")
//    Human().hello()
    //加上？後可以放null值 !!出現null的話例外處理
    var s:String? = "abcde"
    //字串長度值
    println(s?.length)
    //取第幾個字串
    println(s?.get(3))
    //取幾個字串之後
    println(s?.substring(3))

    val h = Human(weight = 66.5f, height = 1.7f); //不能再改變的變數
    println(h.bmi());
    val score = 88;
    val c:Char = 'A'
    println(score > 90)
    println(c.toInt() > 60)

//    h.hello()
//    var age2 = 19; //變數
//    age2 = age2.plus(other = 1); //增加 1
//    val age = 19;  //不可改變的值
//    var weight = 66.5;
//    var name: String;
//    name = "Andy"
//    print(age2)
}

class Human(var name:String= "", var weight:Float, var height:Float) {
    init {
        println("test $weight")
    }
//    constructor(name:String,weight: Float,height: Float) :this(weight, height)

    fun bmi() :Float {
        var bmi = weight / (height*height)
        return bmi
    }

    fun hello() {
        println("Hello kotlin")
    }
}